plonk-napi/src/lib.rs

```diff
@@ -1,23 +1,50 @@
 mod build_info;
 mod circuit;
+mod gate_vector;
+mod pasta_fp_plonk_index;
+mod pasta_fq_plonk_index;
 mod plonk_verifier_index;
 mod poly_comm;
 mod poseidon;
-mod prover_index;
-mod types;
+mod srs;
+mod tables;
 mod vector;
 mod wrappers;
 
 pub use circuit::prover_to_json;
+pub use gate_vector::{
+    caml_pasta_fp_plonk_circuit_serialize, caml_pasta_fp_plonk_gate_vector_add,
+    caml_pasta_fp_plonk_gate_vector_create, caml_pasta_fp_plonk_gate_vector_digest,
+    caml_pasta_fp_plonk_gate_vector_from_bytes, caml_pasta_fp_plonk_gate_vector_get,
+    caml_pasta_fp_plonk_gate_vector_len, caml_pasta_fp_plonk_gate_vector_to_bytes,
+    caml_pasta_fp_plonk_gate_vector_wrap, caml_pasta_fq_plonk_circuit_serialize,
+    caml_pasta_fq_plonk_gate_vector_add, caml_pasta_fq_plonk_gate_vector_create,
+    caml_pasta_fq_plonk_gate_vector_digest, caml_pasta_fq_plonk_gate_vector_from_bytes,
+    caml_pasta_fq_plonk_gate_vector_get, caml_pasta_fq_plonk_gate_vector_len,
+    caml_pasta_fq_plonk_gate_vector_to_bytes, caml_pasta_fq_plonk_gate_vector_wrap,
+    NapiFpGate as WasmFpGate, NapiFpGateVector as WasmFpGateVector, NapiFqGate as WasmFqGate,
```

Comment:
this was actually a little confusing when i worked on top of this PR - I think we should probbably leave it as `NapiXX` and leave it to the crate that imports these to rename them if needed, wdyt? we cna do this in a follow up PR tho